sourmash/lca/lca_utils.py

```diff
@@ -382,7 +382,11 @@ def _signatures(self):
 
         for (k, v) in self.hashval_to_idx.items():
             for vv in v:
-                temp_vals[vv].append(k)
+                temp_hashes = temp_vals[vv]
+                temp_hashes.append(k)
+                if len(temp_hashes) > 50:
```

Comment:
I chose `50` here, but it's pretty arbitrary. If you really want to micro-optimize, `list` is reiszed and grow in [this pattern]: `0, 4, 8, 16, 25, 35, 46, 58, 72, 88, ...`

[this pattern]: https://github.com/python/cpython/blob/b2b4a51f7463a0392456f7772f33223e57fa4ccc/Objects/listobject.c#L57